### ISSUE TITLE
latex_cwl_completions Error Message

### ISSUE THREAD

**Issue by TheGraviton** [SublimeText/LaTeXTools#933]
I'm running Sublime Text 2 on Mac OS 10.12.1. I've been getting the following error message every few days for several weeks: "A plugin (latex_cwl_completions) may be making Sublime Test unresponsive by taking too long (0.338010s) in its on_activated callback."

It then tells me that I can disable this error message in the user preferences, but I'd rather just fix the error itself. Any ideas?

**Comment by r-stein** [comment]
Do you use cwl completions to auto complete LaTeX commands? The latex_cwl_completions does the most work in a different thread, but has some work (on ST2) on the main thread.

**Comment by TheGraviton** [comment]
Update: I just got the same error message, but for a different plugin: "A plugin (latexing.listener) may be making Sublime Text unresponsive by taking too long (0.218242s) in its on_post_save callback." I guess this isn't specific to one plugin. I currently have three Sublime Text windows with a total of 7 files open. Is that contributing to the error message?

**Comment by TheGraviton** [comment]
r-stein, I'm not sure if I use cwl completions. I'm just using the default setup, which does automatically complete things like quotation marks and braces. On a completely different note, is there a way for me to edit which things get autocompleted? Sometimes I want to write a grave accent mark (the key above tab), and the program always autocompletes the second mark (an apostrophe). I've had to actually copy and paste the accent mark from other text.

**Comment by r-stein** [comment]
No the `on_activated` is just called once. The `on_load` is called for ever file.

**Comment by r-stein** [comment]
The cwl completions completes LaTeX package commands for your used packages und requires the [LaTeX-cwl](https://packagecontrol.io/packages/LaTeX-cwl) package. It will also add your self-defined commands:

![self-defined commands](https://cloud.githubusercontent.com/assets/12573621/11513807/7aecc68c-9875-11e5-89ce-7c002103dba0.gif)

You can just edit your keybindings to overwrite the completions (open KeyBindings - User) and add:

``` js
    {
        "keys": ["`"], "command": "insert", "args": {"characters": "`"},
        "context":
        [
            { "key": "selector", "operator": "equal", "operand": "text.tex.latex" }
        ]
    },
```

**Comment by TheGraviton** [comment]
Thanks. I was able to overwrite that particular completion. But is there anything I can do about the errors?

**Comment by r-stein** [comment]
If you don't use the cwl package you can disable it in the [settings](https://github.com/SublimeText/LaTeXTools/blob/v3.12.7/LaTeXTools.sublime-settings#L64) by setting `command_completion` to `"never"`. This will avoid the check, whether you have cwl installed.
You may also consider switching to ST3, because it has some performance improvements (e.g. those checks are done async).

**Comment by ig0774** [comment]
@TheGraviton: If you're willing to modify the Python code, you can change line 329 of `latex_cwl_completions.py` to `is_cwl_available()` which should reduce the likelihood of this happening.

I'm going to look into other things we could do to make cwl loading more efficient.

**Comment by ig0774** [comment]
This should be fixed in v3.12.9.
